<a href="https://colab.research.google.com/github/siva-97/Neuralnetworks/blob/master/BERT1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!git clone https://github.com/google-research/bert.git

Cloning into 'bert'...
remote: Enumerating objects: 340, done.
remote: Total 340 (delta 0), reused 0 (delta 0), pack-reused 340
Receiving objects: 100% (340/340), 300.28 KiB | 1.47 MiB/s, done.
Resolving deltas: 100% (185/185), done.


In [5]:
!git clone https://github.com/see--/natural-question-answering.git

fatal: destination path 'natural-question-answering' already exists and is not an empty directory.


In [0]:
!pip install -q kaggle

In [7]:
!mkdir ~/.kaggle

mkdir: cannot create directory ‘/root/.kaggle’: File exists


In [0]:
!cp kaggle.json ~/.kaggle/

In [0]:
!chmod 600 ~/.kaggle/kaggle.json

In [10]:
!kaggle competitions download -c tensorflow2-question-answering

  0% 0.00/18.2k [00:00<?, ?B/s]
100% 18.2k/18.2k [00:00<00:00, 33.5MB/s]
100% 4.46G/4.46G [01:30<00:00, 90.4MB/s]
100% 4.46G/4.46G [01:30<00:00, 53.1MB/s]
100% 4.78M/4.78M [00:00<00:00, 31.3MB/s]



In [0]:
cd bert

/content/bert


In [11]:
!unzip /content/simplified-nq-train.jsonl.zip

Archive:  /content/simplified-nq-train.jsonl.zip
  inflating: simplified-nq-train.jsonl  


In [0]:
!unzip /content/uncased_L-12_H-768_A-12.zip

Archive:  /content/uncased_L-12_H-768_A-12.zip
   creating: uncased_L-12_H-768_A-12/
  inflating: uncased_L-12_H-768_A-12/bert_model.ckpt.meta  
  inflating: uncased_L-12_H-768_A-12/bert_model.ckpt.data-00000-of-00001  
  inflating: uncased_L-12_H-768_A-12/vocab.txt  
  inflating: uncased_L-12_H-768_A-12/bert_model.ckpt.index  
  inflating: uncased_L-12_H-768_A-12/bert_config.json  


In [0]:
import tokenization

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or 

In [0]:
vocab_file=tokenization.load_vocab('/content/bert/uncased_L-12_H-768_A-12/vocab.txt')

In [0]:
word_piece_tokenizer= tokenization.WordpieceTokenizer(vocab_file)


In [0]:
basic_tokenizer= tokenization.BasicTokenizer()

In [0]:
basic_tokens=basic_tokenizer.tokenize("Hi, How are you embedding?")

In [0]:
basic_tokens

['hi', ',', 'how', 'are', 'you', 'embedding', '?']

In [0]:
word_piece_tokens=[]
for token in basic_tokens:
  word_piece_tokens.append(word_piece_tokenizer.tokenize(token))

In [0]:
word_piece_tokens

[['hi'], [','], ['how'], ['are'], ['you'], ['em', '##bed', '##ding'], ['?']]

In [0]:
word_piece_tokenizer.tokenize('embedding')

['em', '##bed', '##ding']

In [0]:
tokens_to_ids=[]
for word in word_piece_tokens:
  tokens_to_ids.append(tokenization.convert_tokens_to_ids(vocab=vocab_file,tokens=word))

In [0]:
tokens_to_ids

[[7632], [1010], [2129], [2024], [2017], [7861, 8270, 4667], [1029]]

In [0]:
import modeling

In [0]:
tokens_to_ids

[[7632], [1010], [2129], [2024], [2017], [7861, 8270, 4667], [1029]]

In [0]:
import run_squad

In [0]:
!export BERT_BASE_DIR=/content/bert/uncased_L-12_H-768_A-12/
!export SQUAD_DIR=/content/
!cd $BERT_BASE_DIR
!pwd

/content/bert


In [13]:
!python /content/natural-question-answering/nq_to_squad.py --fn='/content/simplified-nq-train.jsonl' --val_ids='/content/natural-question-answering/val_ids.csv' --num_samples=15000

Converting /content/simplified-nq-train.jsonl to nq-train-v1.0.1.json & nq-val-v1.0.1.json ... 
  ------------ STATS ------------------
  Found 185 yes/no, 318 very long and 121 short of 15000 and trimmed 0
  #short 5235 #long 7351 of 14815
Wrote 13375 entries to nq-train-v1.0.1.json
Wrote 1440 entries to nq-val-v1.0.1.json
100% 14962/15000 [00:25<00:00, 637.07it/s, ltotal=8947, llong=179, long_end=1423.45]Wrote csv to nq-val-v1.0.1.csv
Using val ids from: /content/natural-question-answering/val_ids.csv
100% 14962/15000 [00:26<00:00, 567.97it/s, ltotal=8947, llong=179, long_end=1423.45]


In [15]:
cd /content/natural-question-answering/transformers_repo/

/content/natural-question-answering/transformers_repo


In [16]:
!sudo python3 /content/natural-question-answering/transformers_repo/setup.py install

running install
running bdist_egg
running egg_info
creating src/transformers.egg-info
writing src/transformers.egg-info/PKG-INFO
writing dependency_links to src/transformers.egg-info/dependency_links.txt
writing requirements to src/transformers.egg-info/requires.txt
writing top-level names to src/transformers.egg-info/top_level.txt
writing manifest file 'src/transformers.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/transformers.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib
creating build/lib/transformers
copying src/transformers/modeling_auto.py -> build/lib/transformers
copying src/transformers/modeling_xlm_roberta.py -> build/lib/transformers
copying src/transformers/tokenization_ctrl.py -> build/lib/transformers
copying src/transformers/pipelines.py -> build/lib/transformers
copying src/transformers/modeling_encoder_decoder.py -> build/li

In [17]:
cd /content/natural-question-answering/

/content/natural-question-answering


In [18]:
!mv /content/nq-train-v1.0.1.json /content/natural-question-answering/
!mv /content/nq-val-v1.0.1.json /content/natural-question-answering/

ERROR! Session/line number was not unique in database. History logging moved to new session 59


In [19]:
cd /content/natural-question-answering/

/content/natural-question-answering


In [20]:

!python3  /content/natural-question-answering/train_eval.py \
  --train_fn='/content/natural-question-answering/nq-train-v1.0.1.json' \
  --predict_fn='/content/natural-question-answering/sample_eval.json' \


Streaming output truncated to the last 5000 lines.
2020-04-05 08:47:43.626410: I tensorflow/core/common_runtime/bfc_allocator.cc:990] InUse at 7f5055400000 of size 16777216 next 1343
2020-04-05 08:47:43.626421: I tensorflow/core/common_runtime/bfc_allocator.cc:990] InUse at 7f5056400000 of size 146800640 next 1321
2020-04-05 08:47:43.626432: I tensorflow/core/common_runtime/bfc_allocator.cc:990] InUse at 7f505f000000 of size 16384 next 1359
2020-04-05 08:47:43.626442: I tensorflow/core/common_runtime/bfc_allocator.cc:990] InUse at 7f505f004000 of size 16384 next 1360
2020-04-05 08:47:43.626452: I tensorflow/core/common_runtime/bfc_allocator.cc:990] InUse at 7f505f008000 of size 16384 next 1371
2020-04-05 08:47:43.626462: I tensorflow/core/common_runtime/bfc_allocator.cc:990] InUse at 7f505f00c000 of size 16384 next 1382
2020-04-05 08:47:43.626472: I tensorflow/core/common_runtime/bfc_allocator.cc:990] InUse at 7f505f010000 of size 16384 next 1443
2020-04-05 08:47:43.626481: I tensorflo

INFO:tensorflow:*** Example ***
I0404 22:09:11.290320 140039054059392 run_squad.py:431] *** Example ***
INFO:tensorflow:unique_id: 1000000017
I0404 22:09:11.290471 140039054059392 run_squad.py:432] unique_id: 1000000017
INFO:tensorflow:example_index: 17
I0404 22:09:11.290569 140039054059392 run_squad.py:433] example_index: 17
INFO:tensorflow:doc_span_index: 0
I0404 22:09:11.290659 140039054059392 run_squad.py:434] doc_span_index: 0
INFO:tensorflow:tokens: [CLS] what other beginnings of origin ##ation do some of the last names of the greeks share ? [SEP] greek surname ##s were widely in use by the 9th century su ##pp ##lan ##ting the ancient tradition of using the father ’ s name , however greek surname ##s are most commonly patron ##ym ##ics . commonly , greek male surname ##s end in - s , which is the common ending for greek masculine proper nouns in the no ##mina ##tive case . exceptionally , some end in - ou , indicating the gen ##itive case of this proper noun for patron ##ym ##ic reasons . although surname ##s in mainland greece are static today , dynamic and changing patron ##ym ##ic usage survives in middle names where the gen ##itive of father ' s first name is commonly the middle name ( this usage having been passed on to the russians ) . in cyprus , by contrast , surname ##s follow the ancient tradition of being given according to the father ’ s name . finally , in addition to greek - derived surname ##s many have latin , turkish and italian origin . [SEP]
I0404 22:09:11.290821 140039054059392 run_squad.py:436] tokens: [CLS] what other beginnings of origin ##ation do some of the last names of the greeks share ? [SEP] greek surname ##s were widely in use by the 9th century su ##pp ##lan ##ting the ancient tradition of using the father ’ s name , however greek surname ##s are most commonly patron ##ym ##ics . commonly , greek male surname ##s end in - s , which is the common ending for greek masculine proper nouns in the no ##mina ##tive case . exceptionally , some end in - ou , indicating the gen ##itive case of this proper noun for patron ##ym ##ic reasons . although surname ##s in mainland greece are static today , dynamic and changing patron ##ym ##ic usage survives in middle names where the gen ##itive of father ' s first name is commonly the middle name ( this usage having been passed on to the russians ) . in cyprus , by contrast , surname ##s follow the ancient tradition of being given according to the father ’ s name . finally , in addition to greek - derived surname ##s many have latin , turkish and italian origin . [SEP]
INFO:tensorflow:token_to_orig_map: 19:0 20:1 21:1 22:2 23:3 24:4 25:5 26:6 27:7 28:8 29:9 30:10 31:10 32:10 33:10 34:11 35:12 36:13 37:14 38:15 39:16 40:17 41:17 42:17 43:18 44:18 45:19 46:20 47:21 48:21 49:22 50:23 51:24 52:25 53:25 54:25 55:25 56:26 57:26 58:27 59:28 60:29 61:29 62:30 63:31 64:32 65:32 66:32 67:33 68:34 69:35 70:36 71:37 72:38 73:39 74:40 75:41 76:42 77:43 78:44 79:45 80:45 81:45 82:46 83:46 84:47 85:47 86:48 87:49 88:50 89:51 90:51 91:51 92:52 93:53 94:54 95:54 96:55 97:56 98:57 99:58 100:59 101:60 102:61 103:61 104:61 105:62 106:62 107:63 108:64 109:64 110:65 111:66 112:67 113:68 114:69 115:70 116:70 117:71 118:72 119:73 120:74 121:74 122:74 123:75 124:76 125:77 126:78 127:79 128:80 129:81 130:82 131:82 132:83 133:84 134:84 135:84 136:85 137:86 138:87 139:88 140:89 141:90 142:91 143:92 144:92 145:93 146:94 147:95 148:96 149:97 150:98 151:99 152:100 153:100 154:100 155:101 156:102 157:102 158:103 159:104 160:104 161:105 162:105 163:106 164:107 165:108 166:109 167:110 168:111 169:112 170:113 171:114 172:115 173:116 174:116 175:116 176:117 177:117 178:118 179:118 180:119 181:120 182:121 183:122 184:122 185:122 186:123 187:123 188:124 189:125 190:126 191:126 192:127 193:128 194:129 195:130 196:130
I0404 22:09:11.290992 140039054059392 run_squad.py:438] token_to_orig_map: 19:0 20:1 21:1 22:2 23:3 24:4 25:5 26:6 27:7 28:8 29:9 30:10 31:10 32:10 33:10 34:11 35:12 36:13 37:14 38:15 39:16 40:17 41:17 42:17 43:18 44:18 45:19 46:20 47:21 48:21 49:22 50:23 51:24 52:25 53:25 54:25 55:25 56:26 57:26 58:27 59:28 60:29 61:29 62:30 63:31 64:32 65:32 66:32 67:33 68:34 69:35 70:36 71:37 72:38 73:39 74:40 75:41 76:42 77:43 78:44 79:45 80:45 81:45 82:46 83:46 84:47 85:47 86:48 87:49 88:50 89:51 90:51 91:51 92:52 93:53 94:54 95:54 96:55 97:56 98:57 99:58 100:59 101:60 102:61 103:61 104:61 105:62 106:62 107:63 108:64 109:64 110:65 111:66 112:67 113:68 114:69 115:70 116:70 117:71 118:72 119:73 120:74 121:74 122:74 123:75 124:76 125:77 126:78 127:79 128:80 129:81 130:82 131:82 132:83 133:84 134:84 135:84 136:85 137:86 138:87 139:88 140:89 141:90 142:91 143:92 144:92 145:93 146:94 147:95 148:96 149:97 150:98 151:99 152:100 153:100 154:100 155:101 156:102 157:102 158:103 159:104 160:104 161:105 162:105 163:106 164:107 165:108 166:109 167:110 168:111 169:112 170:113 171:114 172:115 173:116 174:116 175:116 176:117 177:117 178:118 179:118 180:119 181:120 182:121 183:122 184:122 185:122 186:123 187:123 188:124 189:125 190:126 191:126 192:127 193:128 194:129 195:130 196:130
INFO:tensorflow:token_is_max_context: 19:True 20:True 21:True 22:True 23:True 24:True 25:True 26:True 27:True 28:True 29:True 30:True 31:True 32:True 33:True 34:True 35:True 36:True 37:True 38:True 39:True 40:True 41:True 42:True 43:True 44:True 45:True 46:True 47:True 48:True 49:True 50:True 51:True 52:True 53:True 54:True 55:True 56:True 57:True 58:True 59:True 60:True 61:True 62:True 63:True 64:True 65:True 66:True 67:True 68:True 69:True 70:True 71:True 72:True 73:True 74:True 75:True 76:True 77:True 78:True 79:True 80:True 81:True 82:True 83:True 84:True 85:True 86:True 87:True 88:True 89:True 90:True 91:True 92:True 93:True 94:True 95:True 96:True 97:True 98:True 99:True 100:True 101:True 102:True 103:True 104:True 105:True 106:True 107:True 108:True 109:True 110:True 111:True 112:True 113:True 114:True 115:True 116:True 117:True 118:True 119:True 120:True 121:True 122:True 123:True 124:True 125:True 126:True 127:True 128:True 129:True 130:True 131:True 132:True 133:True 134:True 135:True 136:True 137:True 138:True 139:True 140:True 141:True 142:True 143:True 144:True 145:True 146:True 147:True 148:True 149:True 150:True 151:True 152:True 153:True 154:True 155:True 156:True 157:True 158:True 159:True 160:True 161:True 162:True 163:True 164:True 165:True 166:True 167:True 168:True 169:True 170:True 171:True 172:True 173:True 174:True 175:True 176:True 177:True 178:True 179:True 180:True 181:True 182:True 183:True 184:True 185:True 186:True 187:True 188:True 189:True 190:True 191:True 192:True 193:True 194:True 195:True 196:True
I0404 22:09:11.291152 140039054059392 run_squad.py:440] token_is_max_context: 19:True 20:True 21:True 22:True 23:True 24:True 25:True 26:True 27:True 28:True 29:True 30:True 31:True 32:True 33:True 34:True 35:True 36:True 37:True 38:True 39:True 40:True 41:True 42:True 43:True 44:True 45:True 46:True 47:True 48:True 49:True 50:True 51:True 52:True 53:True 54:True 55:True 56:True 57:True 58:True 59:True 60:True 61:True 62:True 63:True 64:True 65:True 66:True 67:True 68:True 69:True 70:True 71:True 72:True 73:True 74:True 75:True 76:True 77:True 78:True 79:True 80:True 81:True 82:True 83:True 84:True 85:True 86:True 87:True 88:True 89:True 90:True 91:True 92:True 93:True 94:True 95:True 96:True 97:True 98:True 99:True 100:True 101:True 102:True 103:True 104:True 105:True 106:True 107:True 108:True 109:True 110:True 111:True 112:True 113:True 114:True 115:True 116:True 117:True 118:True 119:True 120:True 121:True 122:True 123:True 124:True 125:True 126:True 127:True 128:True 129:True 130:True 131:True 132:True 133:True 134:True 135:True 136:True 137:True 138:True 139:True 140:True 141:True 142:True 143:True 144:True 145:True 146:True 147:True 148:True 149:True 150:True 151:True 152:True 153:True 154:True 155:True 156:True 157:True 158:True 159:True 160:True 161:True 162:True 163:True 164:True 165:True 166:True 167:True 168:True 169:True 170:True 171:True 172:True 173:True 174:True 175:True 176:True 177:True 178:True 179:True 180:True 181:True 182:True 183:True 184:True 185:True 186:True 187:True 188:True 189:True 190:True 191:True 192:True 193:True 194:True 195:True 196:True
INFO:tensorflow:input_ids: 101 2054 2060 16508 1997 4761 3370 2079 2070 1997 1996 2197 3415 1997 1996 13176 3745 1029 102 3306 11988 2015 2020 4235 1999 2224 2011 1996 6280 2301 10514 9397 5802 3436 1996 3418 4535 1997 2478 1996 2269 1521 1055 2171 1010 2174 3306 11988 2015 2024 2087 4141 9161 24335 6558 1012 4141 1010 3306 3287 11988 2015 2203 1999 1011 1055 1010 2029 2003 1996 2691 4566 2005 3306 14818 5372 19211 1999 1996 2053 22311 6024 2553 1012 17077 1010 2070 2203 1999 1011 15068 1010 8131 1996 8991 13043 2553 1997 2023 5372 15156 2005 9161 24335 2594 4436 1012 2348 11988 2015 1999 8240 5483 2024 10763 2651 1010 8790 1998 5278 9161 24335 2594 8192 13655 1999 2690 3415 2073 1996 8991 13043 1997 2269 1005 1055 2034 2171 2003 4141 1996 2690 2171 1006 2023 8192 2383 2042 2979 2006 2000 1996 12513 1007 1012 1999 9719 1010 2011 5688 1010 11988 2015 3582 1996 3418 4535 1997 2108 2445 2429 2000 1996 2269 1521 1055 2171 1012 2633 1010 1999 2804 2000 3306 1011 5173 11988 2015 2116 2031 3763 1010 5037 1998 3059 4761 1012 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
I0404 22:09:11.384069 140039054059392 run_squad.py:442] input_ids: 101 2054 2060 16508 1997 4761 3370 2079 2070 1997 1996 2197 3415 1997 1996 13176 3745 1029 102 3306 11988 2015 2020 4235 1999 2224 2011 1996 6280 2301 10514 9397 5802 3436 1996 3418 4535 1997 2478 1996 2269 1521 1055 2171 1010 2174 3306 11988 2015 2024 2087 4141 9161 24335 6558 1012 4141 1010 3306 3287 11988 2015 2203 1999 1011 1055 1010 2029 2003 1996 2691 4566 2005 3306 14818 5372 19211 1999 1996 2053 22311 6024 2553 1012 17077 1010 2070 2203 1999 1011 15068 1010 8131 1996 8991 13043 2553 1997 2023 5372 15156 2005 9161 24335 2594 4436 1012 2348 11988 2015 1999 8240 5483 2024 10763 2651 1010 8790 1998 5278 9161 24335 2594 8192 13655 1999 2690 3415 2073 1996 8991 13043 1997 2269 1005 1055 2034 2171 2003 4141 1996 2690 2171 1006 2023 8192 2383 2042 2979 2006 2000 1996 12513 1007 1012 1999 9719 1010 2011 5688 1010 11988 2015 3582 1996 3418 4535 1997 2108 2445 2429 2000 1996 2269 1521 1055 2171 1012 2633 1010 1999 2804 2000 3306 1011 5173 11988 2015 2116 2031 3763 1010 5037 1998 3059 4761 1012 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
I0404 22:09:11.384458 140039054059392 run_squad.py:444] input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
I0404 22:09:11.384761 140039054059392 run_squad.py:446] segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
INFO:tensorflow:start_position: 188
I0404 22:09:11.385058 140039054059392 run_squad.py:451] start_position: 188
INFO:tensorflow:end_position: 196
I0404 22:09:11.385160 140039054059392 run_squad.py:452] end_position: 196
INFO:tensorflow:answer: many have latin , turkish and italian origin .
I0404 22:09:11.385274 140039054059392 run_squad.py:454] answer: many have latin , turkish and italian origin .
INFO:tensorflow:*** Example ***

In [0]:
tf.__version__

'1.14.0'

In [0]:
!python run_squad.py \
  --vocab_file=/content/bert/uncased_L-12_H-768_A-12/vocab.txt \
  --bert_config_file=/content/bert/uncased_L-12_H-768_A-12/bert_config.json \
  --init_checkpoint=/content/bert/uncased_L-12_H-768_A-12/bert_model.ckpt \
  --do_train=True \
  --train_file=/content/train-v2.0.json \
  --do_predict=True \
  --predict_file=/content/sample_eval.json \
  --train_batch_size=24 \
  --learning_rate=3e-5 \
  --num_train_epochs=2.0 \
  --max_seq_length=384 \
  --doc_stride=128 \
  --output_dir=/content \
  --use_tpu=False \
  --version_2_with_negative=True

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or 

In [0]:
import run_squad

In [0]:
sample_eval_examples=run_squad.read_squad_examples('/content/sample_eval.json',False)

In [0]:
sample_eval_examples

[qas_id: 56ddde6b9a695914005b9628, question_text: how many units are needed to graduate?, doc_tokens: [Every graduate student should complete 30 units to graduate successfully.The number of units is mandatory and only under special circumstances the load may be reduced. Graduate students can choose 30 units from their own area of specialisation. 6 units can be selected from other departments with special permission from the graduate advisor]]

In [0]:
eval_features=[]
filename='eval_feature.tf_record'
eval_writer=run_squad.FeatureWriter(filename=filename,is_training=False)
def append_feature(feature):
  eval_features.append(feature)
  eval_writer.process_feature(feature)

run_squad.convert_examples_to_features(sample_eval_examples,tokenization.FullTokenizer('/content/bert/uncased_L-12_H-768_A-12/vocab.txt'),512,128,30,False,output_fn=append_feature)

ERROR! Session/line number was not unique in database. History logging moved to new session 67
INFO:tensorflow:*** Example ***
INFO:tensorflow:unique_id: 1000000000
INFO:tensorflow:example_index: 0
INFO:tensorflow:doc_span_index: 0
INFO:tensorflow:tokens: [CLS] how many units are needed to graduate ? [SEP] every graduate student should complete 30 units to graduate successfully . the number of units is mandatory and only under special circumstances the load may be reduced . graduate students can choose 30 units from their own area of special ##isation . 6 units can be selected from other departments with special permission from the graduate advisor [SEP]
INFO:tensorflow:token_to_orig_map: 10:0 11:1 12:2 13:3 14:4 15:5 16:6 17:7 18:8 19:9 20:9 21:9 22:10 23:11 24:12 25:13 26:14 27:15 28:16 29:17 30:18 31:19 32:20 33:21 34:22 35:23 36:24 37:24 38:25 39:26 40:27 41:28 42:29 43:30 44:31 45:32 46:33 47:34 48:35 49:36 50:36 51:36 52:37 53:38 54:39 55:40 56:41 57:42 58:43 59:44 60:45 61:46 62

In [0]:
eval_features[0].tokens

['[CLS]',
 'how',
 'many',
 'units',
 'are',
 'needed',
 'to',
 'graduate',
 '?',
 '[SEP]',
 'every',
 'graduate',
 'student',
 'should',
 'complete',
 '30',
 'units',
 'to',
 'graduate',
 'successfully',
 '.',
 'the',
 'number',
 'of',
 'units',
 'is',
 'mandatory',
 'and',
 'only',
 'under',
 'special',
 'circumstances',
 'the',
 'load',
 'may',
 'be',
 'reduced',
 '.',
 'graduate',
 'students',
 'can',
 'choose',
 '30',
 'units',
 'from',
 'their',
 'own',
 'area',
 'of',
 'special',
 '##isation',
 '.',
 '6',
 'units',
 'can',
 'be',
 'selected',
 'from',
 'other',
 'departments',
 'with',
 'special',
 'permission',
 'from',
 'the',
 'graduate',
 'advisor',
 '[SEP]']

In [0]:
predict_fn=run_squad.input_fn_builder(eval_writer.filename,512,False,False)